azurelinuxagent/ga/exthandlers.py

```diff
@@ -275,6 +275,12 @@ def _incarnation_changed(self, etag):
         # Skip processing if GoalState incarnation did not change
         return self.last_etag != etag
 
+    def get_activity_and_correlation_id(self):
+
+        def check_empty(value): return value if value else "NA"
```

Comment:
let's get in the habit of making explicit comparisons against None ("if value is not None")

Reply:
Done (and will keep in mind from next time)

Reply:
Great! This is a good example of why we need to be explicit. When I read this code I did not realize that empty string was a valid value. Doing 'if value not in (None, '')' this instead would clarify the intention of this code.

Although scripting (and other) languages provide this kind of flexibility, as the size of production code grows, it is better to do explicit checks, rather than debug obscure bugs. In this specific case being explicit vs being the implicit may not make a difference, but in other cases it does. Also, as the code evolves an implicit check that was initially OK may end up being incorrect.

`Though I'll keep in mind to always have a None check if there is only None to be compared against (and not empty strings or empty arrays [])
`

Yes, when checking against None please also always be explicit. PEP 8 touches briefly on this:

`
Also, beware of writing if x when you really mean if x is not None -- e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!`

Reply:
Makes sense, I'll send an update

Reply:
Done